app/controllers/katello/api/registry/registry_proxies_controller.rb

```diff
@@ -293,6 +301,11 @@ def v1_ping
     end
 
     def v1_search
+      if SETTINGS[:katello][:block_registry_v1_search]
+        render json: {}, status: :not_found
+        return
+      end
```

Comment:
I would suggest making this a Setting instead of a SETTING (defining it in app/models/settings/content.rb)   Anything that is a SETTING requires installer updates to be able to consistently set it.

Reply:
Also if its a Setting, i'd probably recommend we change it to be positive instead of negative (allow_v1_docker_search) and default it to true.

Reply:
So while i was looking at this just now, i realized we might could use the HTTP_USER_AGENT to determien whether to throw a 404 or not.  Thoughts?

From docker:
```
172.17.0.1 - - [27/Jul/2020:17:37:50 +0000] "GET /v1/search?q=&n=25 HTTP/1.1" 404 19 "" "docker/1.13.1 go/go1.12.7 kernel/5.6.13-100.fc30.x86_64 os/linux arch/amd64 UpstreamClient(Docker-Client/1.13.1 \\(linux\\))"
```


from podman:
```
172.17.0.1 - - [27/Jul/2020:17:41:54 +0000] "GET /v1/search?n=25&q=foo HTTP/1.1" 404 19 "" "libpod/1.8.0"
```